### PR TITLE
[CPU][ARM][DOC] ARM documentation - 23.1 release

### PR DESCRIPTION
**Pull request by alvoron**

### CODE REVIEW

**Comment by alvoron** [comment]
Master PR: https://github.com/openvinotoolkit/openvino/pull/18895

File: docs/OV_Runtime_UG/supported_plugins/CPU.md [hunk 1]
```diff
@@ -90,9 +90,16 @@ CPU plugin supports the following floating-point data types as inference precisi
 
 - ``f32`` (Intel® x86-64, Arm®)
 - ``bf16`` (Intel® x86-64)
+- ``f16`` (Arm®, experimental feature)
 
-The default floating-point precision of a CPU primitive is ``f32``. To support the ``f16`` OpenVINO IR the plugin internally converts 
+The default floating-point precision of a CPU primitive is ``f32``. By default to support the ``f16`` OpenVINO IR the plugin internally converts 
 all the ``f16`` values to ``f32`` and all the calculations are performed using the native precision of ``f32``.
+On platforms that natively support ``float16`` calculations, the ``float16`` type is used if inferece precision is explicitly set to ``f16`` (see the `Inference Precision Hint <#inference-precision-hint>`__).
```
Comment (tsavina) [inline docs/OV_Runtime_UG/supported_plugins/CPU.md:97]:
```suggestion
On platforms that natively support ``float16`` calculations, the ``float16`` type is used if inference precision is explicitly set to ``f16`` (see the `Inference Precision Hint <#inference-precision-hint>`__).
```

File: docs/OV_Runtime_UG/supported_plugins/CPU.md [hunk 2]
```diff
@@ -90,9 +90,16 @@ CPU plugin supports the following floating-point data types as inference precisi
 
 - ``f32`` (Intel® x86-64, Arm®)
 - ``bf16`` (Intel® x86-64)
+- ``f16`` (Arm®, experimental feature)
 
-The default floating-point precision of a CPU primitive is ``f32``. To support the ``f16`` OpenVINO IR the plugin internally converts 
+The default floating-point precision of a CPU primitive is ``f32``. By default to support the ``f16`` OpenVINO IR the plugin internally converts
```
Comment (tsavina) [inline docs/OV_Runtime_UG/supported_plugins/CPU.md:95]:
```suggestion
The default floating-point precision of a CPU primitive is ``f32``. By default, to support the ``f16`` OpenVINO IR the plugin internally converts 
```

**Review by tsavina — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Comment by alvoron** [comment]
@tsavina the PR has been approved by Dmitry, CPU team is OK to merge it.

**Pull request merged**